iOS/PartySample/app/src/SimpleClient.mm

```diff
@@ -330,7 +362,114 @@ +(void)globalShutdown
                        encoding : CFStringConvertEncodingToNSStringEncoding ( kCFStringEncodingUTF8 ) ];
   return pString;
 }
-                   
+
+#pragma mark - ChatEventHandler
+-(void) onNetworkCreated:(NSString*) networkDescriptor
+{
+    self.networkDescriptor = networkDescriptor;
+
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self->_chatEventHandler onNetworkCreated:networkDescriptor];
+    });
+}
+
+-(void) onGetDescriptorCompleted:(NSString*) networkId withNetworkDescriptor:(NSString*) networkDescriptor
+{
+    self.networkId = networkId;
+    self.networkDescriptor = networkDescriptor;
+
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        m_impl->ConnectToNetwork(NSStringToString(networkId), NSStringToString(networkDescriptor), false);
+    });
+}
+
+-(void) onJoinedNetwork
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self->_chatEventHandler onJoinedNetwork];
+    });
+}
+
+-(void) onPlayerJoin:(NSString*) userId
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self->_chatEventHandler onPlayerJoin:userId];
+    });
+}
+
+-(void) onPlayerLeft:(NSString*) userId
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self->_chatEventHandler onPlayerLeft:userId];
+    });
+}
+
+-(void) onPlayerStatusUpdateStart
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self->_chatEventHandler onPlayerStatusUpdateStart];
+    });
+}
+
+-(void) onPlayerStatusChange:(NSString*)playerId withIndicator:(NSString*)indicatorName
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self->_chatEventHandler onPlayerStatusChange:playerId withIndicator:indicatorName];
+    });
+}
+
+-(void) onPlayerStatusUpdateEnd
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self->_chatEventHandler onPlayerStatusUpdateEnd];
+    });
+}
+
+-(void) onTextMessageReceived:(NSString*) userId withText:(NSString*) text isTranscription:(BOOL) isTranscription
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self->_chatEventHandler onTextMessageReceived:userId withText:text isTranscription:isTranscription];
+    });
+}
+
+-(void) onToastMessage:(NSString*) text
+{
+    [_chatEventHandler onToastMessage:text];
+}
+
+-(void) onStartLoading
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self->_chatEventHandler onStartLoading];
+    });
+}
+
+-(void) onEndLoading
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self->_chatEventHandler onEndLoading];
+    });
+}
+
+#pragma mark - NotificationReceiver
+- (void)didEnterBackground
```

Comment:
>  [](http://example.com/codeflow?start=1&length=1)

nit: for consistency we can remove the extra spaces as in other places we don't have them